jenkins/opensearch-dashboards/distribution-build.jenkinsfile

```diff
@@ -872,44 +872,55 @@ pipeline {
                         }
                     }
                 }
-            }
-        }
-        stage('docker build') {
-            when {
-                beforeAgent true
-                allOf {
-                    expression {
-                        params.BUILD_DOCKER != 'do_not_build_docker'
+                stage('docker build') {
+                    when {
+                        beforeAgent true
+                        allOf {
+                            expression {
+                                params.BUILD_DOCKER != 'do_not_build_docker'
+                            }
+                            expression {
+                                params.BUILD_PLATFORM.contains('linux')
+                            }
+                            expression {
+                                params.BUILD_DISTRIBUTION.contains('tar')
+                            }
+                        }
                     }
-                    expression {
-                        params.BUILD_PLATFORM.contains('linux')
+                    options {
+                        timeout(time: 90, unit: 'MINUTES')
                     }
-                    expression {
-                        params.BUILD_DISTRIBUTION.contains('tar')
+                    agent {
+                        docker {
+                            label AGENT_LINUX_X64
+                            image dockerAgent.image
+                            args dockerAgent.args
+                            registryUrl 'https://public.ecr.aws/'
+                            alwaysPull true
+                        }
                     }
-                }
-            }
-            agent {
-                docker {
-                    label AGENT_LINUX_X64
-                    image dockerAgent.image
-                    args dockerAgent.args
-                    registryUrl 'https://public.ecr.aws/'
-                    alwaysPull true
-                }
-            }
-            steps {
-                script {
-                    echo "env.ARTIFACT_URL_LINUX_X64_TAR: ${env.ARTIFACT_URL_LINUX_X64_TAR}"
-                    echo "env.ARTIFACT_URL_LINUX_ARM64_TAR: ${env.ARTIFACT_URL_LINUX_ARM64_TAR}"
+                    steps {
+                        script {
+                            while(true) {
```

Comment:
Would recommend to use inbuilt feature of [waitUntil](https://www.jenkins.io/doc/pipeline/steps/workflow-basic-steps/#waituntil-wait-for-condition). We can take up it up as an enhancement if you like. Would avoid us adding code for checking and waiting for 120seconds